flink-runtime/src/main/java/org/apache/flink/runtime/instance/SimpleSlot.java

```diff
@@ -37,16 +41,18 @@
  * <p>If this slot is part of a {@link SharedSlot}, then the parent attribute will point to that shared slot.
  * If not, then the parent attribute is null.
  */
-public class SimpleSlot extends Slot {
+public class SimpleSlot extends Slot implements LogicalSlot {
 
 	/** The updater used to atomically swap in the execution */
```

Comment:
Will the payload always be the `execution`?

Reply:
Effectively yes. Will replace execution by payload.